extensions/ql-vscode/src/remote-queries/repository-selection.ts

```diff
@@ -41,20 +42,32 @@ export async function getRepositorySelection(
     if (selectedDbItem) {
       switch (selectedDbItem.kind) {
         case DbItemKind.LocalDatabase || DbItemKind.LocalList:
-          throw new Error("Local databases and lists are not supported yet.");
+          void showAndLogErrorMessage(
+            "Local databases and lists are not supported yet.",
+          );
+          break;
         case DbItemKind.RemoteSystemDefinedList:
           return { repositoryLists: [selectedDbItem.listName] };
         case DbItemKind.RemoteUserDefinedList:
-          return {
-            repositories: selectedDbItem.repos.map((repo) => repo.repoFullName),
-          };
+          if (selectedDbItem.repos.length === 0) {
+            void showAndLogErrorMessage(
+              "The selected repository list is empty. Please add repositories to it before running a variant analysis.",
+            );
+            break;
+          } else {
+            return {
+              repositories: selectedDbItem.repos.map(
+                (repo) => repo.repoFullName,
+              ),
+            };
+          }
         case DbItemKind.RemoteOwner:
           return { owners: [selectedDbItem.ownerName] };
         case DbItemKind.RemoteRepo:
           return { repositories: [selectedDbItem.repoFullName] };
       }
     } else {
-      throw new Error(
+      void showAndLogErrorMessage(
```

Comment:
I think this makes sense for it to be reported as an unhandled error because it _should_ never happen.